src/codegen/codegen_neuron_cpp_visitor.cpp

```diff
@@ -1775,6 +1799,70 @@ void CodegenNeuronCppVisitor::print_net_event_call(const ast::FunctionCall& node
     throw std::runtime_error("Not implemented.");
 }
 
+/**
+ * Rename arguments to NET_RECEIVE block with corresponding pointer variable
+ *
+ * \code{.mod}
+ *      NET_RECEIVE (weight, R){
+ *            x = R
+ *      }
+ * \endcode
+ *
+ * then generated code should be:
+ *
+ * \code{.cpp}
+ *      x[id] = _args[1];
+ * \endcode
+ *
+ * So, the `R` in AST needs to be renamed with `_args[1]`.
+ */
+static void rename_net_receive_arguments(const ast::NetReceiveBlock& net_receive_node,
+                                         const ast::Node& node) {
+    const auto& parameters = net_receive_node.get_parameters();
+
+    auto n_parameters = parameters.size();
+    for (size_t i = 0; i < n_parameters; ++i) {
+        const auto& name = parameters[i]->get_node_name();
+        auto var_used = VarUsageVisitor().variable_used(node, name);
+        if (var_used) {
+            RenameVisitor vr(name, fmt::format("_args[{}]", i));
+            node.get_statement_block()->visit_children(vr);
+        }
+    }
+}
+
+void CodegenNeuronCppVisitor::print_net_receive() {
+    auto node = info.net_receive_node;
+    if (!node) {
+        return;
+    }
+
+    ParamVector args;
+    args.emplace_back("", "Point_process*", "", "_pnt");
+    args.emplace_back("", "double*", "", "_args");
+    args.emplace_back("", "double", "", "_lflag");
+
+    printer->fmt_push_block("static void nrn_net_receive_{}({})",
+                            info.mod_suffix,
+                            get_parameter_str(args));
+
+    rename_net_receive_arguments(*node, *node);
+
+    printer->add_line("_nrn_mechanism_cache_instance _ml_obj{_pnt->prop};");
+    printer->add_line("auto * _nt = (NrnThread*) _pnt->_vnt;");
```

Comment:
Minor nitpick: shouldn't we use `static_cast` here (just for the sake of explicitness)?